python/taichi/ad/_ad.py

```diff
@@ -205,24 +206,33 @@ def __enter__(self):
 
         # Attach the context manager to runtime
         self.runtime.target_tape = self
+        return self
```

Comment:
why returning self here?

Reply:
This enables `with ti.ad.Tape() as t`, make it possible to call some tape function inside the context manager scope.